src/stream/src/executor/backfill/snapshot_backfill.rs

```diff
@@ -0,0 +1,494 @@
+                            self.rate_limit,
+                            &mut self.barrier_rx,
+                            &self.output_indices,
+                            self.progress,
+                            first_recv_barrier,
+                        );
+
+                        pin_mut!(snapshot_stream);
+
+                        while let Some(message) = with_consuming_upstream(
+                            snapshot_stream.try_next(),
+                            &mut upstream_buffer,
+                        )
+                        .await?
+                        {
+                            yield message;
+                        }
+                    }
+
+                    let recv_barrier = self.barrier_rx.recv().await.expect("should exist");
+                    assert_eq!(first_barrier.epoch, recv_barrier.epoch);
+                    yield Message::Barrier(first_barrier);
+                }
+
+                let mut barrier_epoch = first_barrier_epoch;
+
+                // Phase 2: consume upstream log store
+                while let Some(barrier) = upstream_buffer.take_buffered_barrier().await? {
+                    let recv_barrier = receive_next_barrier(&mut self.barrier_rx).await?;
+                    assert_eq!(barrier.epoch, recv_barrier.epoch);
+                    assert_eq!(barrier_epoch.curr, barrier.epoch.prev);
+                    barrier_epoch = barrier.epoch;
+
+                    debug!(?barrier_epoch, kind = ?barrier.kind, "before consume change log");
+                    let stream = self
+                        .upstream_table
+                        .batch_iter_log_with_pk_bounds(barrier_epoch.prev, barrier_epoch.prev)
+                        .await?;
+                    let data_types = self.upstream_table.schema().data_types();
+                    let builder = create_builder(None, self.chunk_size, data_types);
+                    let stream = read_change_log(stream, builder);
+                    pin_mut!(stream);
+                    while let Some(chunk) =
+                        with_consuming_upstream(stream.try_next(), &mut upstream_buffer).await?
+                    {
+                        debug!(
+                            ?barrier_epoch,
+                            size = chunk.cardinality(),
+                            "consume change log yield chunk",
+                        );
+                        yield Message::Chunk(chunk);
+                    }
+
+                    debug!(?barrier_epoch, "after consume change log");
+
+                    yield Message::Barrier(barrier);
+                }
+            } else {
+                debug!("skip backfill");
+                let first_recv_barrier = receive_next_barrier(&mut self.barrier_rx).await?;
+                assert_eq!(first_barrier.epoch, first_recv_barrier.epoch);
+                yield Message::Barrier(first_barrier);
+            }
+        }
+        // Phase 3: consume upstream
```

Comment:
We can add a trace here to indicate that backfill is done.

Reply:
Do you mean something like an `info` log so that we can know that the backfill is finished from the log? Or there is currently some special trace collection mechanism that leverage a trace here?

Reply:
> Do you mean something like an info log so that we can know that the backfill is finished from the log? Or there is currently some special trace collection mechanism that leverage a trace here?

Sorry I mean an `info` log.